### PR TITLE
[KEYCLOAK-7021] - keycloak-authz.js and keycloak-authz.d.ts do not work with TypeScript

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

File: adapters/oidc/js/src/main/resources/keycloak-authz.d.ts [hunk 1]
```diff
@@ -35,25 +35,88 @@ declare namespace KeycloakAuthorization {
...
+         * rpt parameter, only the last N requested permissions will be kept in the RPT.
+         */
+        response_permissions_limit?:number
+    }
+
+    interface ResourcePermission {
+        /**
+         * The id or name of a resource.
+         */
+        id?:string,
```
Comment (ssilvert) [inline adapters/oidc/js/src/main/resources/keycloak-authz.d.ts]: Just wondering, is the id supposed to be optional?  Seems a little odd to allow something with the id missing.

Comment (pedroigor) [reply adapters/oidc/js/src/main/resources/keycloak-authz.d.ts]: +1. It is not optional. Will fix it.

**Comment by pedroigor** [comment]
@ssilvert, can we merge this one ?

We can work with the update of https://github.com/keycloak/keycloak-js-bower later, I think. Which we need to reference both keycloak and keycloak-auth libraries.

**Review by ssilvert — APPROVED** [review]

**Comment by ssilvert** [comment]
I think it's fine to merge.

**Pull request merged**